src/Authenticator/Result.php

```diff
@@ -44,25 +44,28 @@ class Result implements ResultInterface
     protected $_errors = [];
 
     /**
-     * Sets the result code, identity, and failure messages
+     * Sets the result status, identity, and failure messages
      *
      * @param null|array|\ArrayAccess $data The identity data
-     * @param int $code Error code.
+     * @param string $status Status constant equivalent.
      * @param array $messages Messages.
      * @throws InvalidArgumentException When invalid identity data is passed.
      */
-    public function __construct($data, $code, array $messages = [])
+    public function __construct($data, $status, array $messages = [])
     {
-        if (empty($data) && $code === self::SUCCESS) {
+        if ($status === self::SUCCESS && empty($data)) {
             throw new InvalidArgumentException('Identity data can not be empty with status success.');
         }
-        if ($data !== null && !is_array($data) && !$data instanceof ArrayAccess) {
+        if ($data !== null && !is_array($data) && !($data instanceof ArrayAccess)) {
```

Comment:
This is better, right?

Reply:
Its the same as far as PHP is concerned.

Reply:
`!($data instanceof ArrayAccess)` is the same as `!$data instanceof ArrayAccess`

Reply:
But it is sure more clear to everyone, especially beginners. So :+1: for keeping it for clarity.